plugins/org.preesm.codegen.xtend/xtend-src/org/preesm/codegen/xtend/printer/c/CHardwarePrinter.xtend

```diff
@@ -730,7 +730,22 @@ class CHardwarePrinter extends CPrinter {
 			}
 			i--;
 		}
-
+		
+		//inserting the PAPIFY functions in the right place
+		i = coreLoop.codeElts.size-1;
+		while (i > 0) {
+			// Retrieve the function ID
+			val elt = coreLoop.codeElts.get(i)
+			if (elt instanceof ActorFunctionCall) {
+				//PreesmLogger.getLogger().info("[HARDWARE DEBUG] the only function call is at line " + i);
```

Comment:
Please remove those commented line (there are more in other commits), or replace them with getLogger().log(Level.FINEST, message);